imap_data_access/cli.py

```diff
@@ -99,13 +104,32 @@ def _query_parser(args: argparse.Namespace):
         "version",
         "extension",
     ]
+    latest_version = ""
     query_params = {
         key: value
         for key, value in vars(args).items()
         if key in valid_args and value is not None
     }
+
+    # removing version from query if it is 'latest', updating flag
+    if args.version == "latest":
+        del query_params["version"]
+        latest_version = "latest"
+
     query_results = imap_data_access.query(**query_params)
 
+    # if latest version was included in search then filter returned query for largest.
+    if latest_version == "latest":
```

Comment:
```suggestion
    query_results = imap_data_access.query(**query_params)

    # if latest version was included in search then filter returned query for largest.
    if args.version == "latest":
```

I _think_ you could avoid having to define another variable here by just using `args.version` instead.